internal/executor/instance/persistentworker/isolation/parallels/vm.go

```diff
@@ -0,0 +1,119 @@
+package parallels
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/google/uuid"
+	"strings"
+)
+
+var ErrVMFailed = errors.New("Parallels VM operation failed")
+
+type VM struct {
+	name string
+}
+
+type NetworkAdapterInfo struct {
+	MAC string
+}
+
+type HardwareInfo struct {
+	Net0 NetworkAdapterInfo
+}
+
+type VirtualMachineInfo struct {
+	ID       string
+	Name     string
+	Hardware HardwareInfo
+}
+
+func NewVMClonedFrom(ctx context.Context, vmNameFrom string) (*VM, error) {
+	vm := &VM{
+		name: "cirrus-" + uuid.New().String(),
+	}
+
+	_, stderr, err := Prlctl(ctx, "clone", vmNameFrom, "--name", vm.name)
```

Comment:
What do you think about using a linked clone here? It seems it's pretty instantaneous? Not sure about IO implications of this copy-on-write like thing though...
```suggestion
	_, stderr, err := Prlctl(ctx, "clone", vmNameFrom, "--linked", "--name", vm.name)
```